Jenkinsfile

```diff
@@ -54,5 +55,24 @@ pipeline {
                 archiveArtifacts artifacts: "$APK_PATH/*.apk", onlyIfSuccessful: true
             }
         }
+        stage('checkout test project repo') {
+            steps {
+                sh "mkdir -p $TEST_PROJECT_REPO_NAME"
+                dir("$TEST_PROJECT_REPO_NAME"){
+                    checkout([$class: 'GitSCM', branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[url: 'https://github.com/edx/edx-app-test.git']]])
+                }
+            }
+        }
+        stage('prepare package for aws device farm') {
+            steps {
+                sh 'bash ./resources/prepare_aws_package.sh'
+            }
+        }
+
+        stage('archive test package') {
+            steps {
+                archiveArtifacts artifacts: "$TEST_PROJECT_REPO_NAME/test_bundle.zip", onlyIfSuccessful: true
```

Comment:
We have been including the `archiveArtifacts` command in the stage that creates the artifacts (ex. https://github.com/edx/devstack/blob/master/scripts/Jenkinsfiles/snapshot#L24). This will help the number of stages in your pipeline manageable.

Reply:
done